calour/experiment.py

```diff
@@ -126,8 +126,22 @@ def sparse(self, sparse):
 
     def __repr__(self):
         '''Return a string representation of this object.'''
-        return 'Experiment %s with %d samples, %d features' % (
-            self.description, self.data.shape[0], self.data.shape[1])
+        if self.description:
+            l1 = 'Experiment %s' % self.description
+        else:
+            l1 = 'Experiment'
+        l2 = '-' * len(l1)
+        lines = [l1, l2,
+                 'data dimension: %d samples, %d features' % self.data.shape]
+        try:
+            lines.append('sample IDs: %r' % self.sample_metadata.index)
```

Comment:
maybe add sampleids only if <10? or show S1,S2,S3,...,S8,S9,S10?
otherwise can be very long for real experiment

Reply:
that's done intrinsically by Pandas when we use the DataFrame.index.